ambry-router/src/main/java/com/github/ambry/router/PutOperation.java

```diff
@@ -2089,7 +2101,25 @@ private void finalizeMetadataChunk() {
               passedInBlobProperties.getAccountId(), passedInBlobProperties.getContainerId(),
               passedInBlobProperties.isEncrypted(), passedInBlobProperties.getExternalAssetTag(),
               passedInBlobProperties.getContentEncoding(), passedInBlobProperties.getFilename(), null);
-      if (isStitchOperation() || getNumDataChunks() > 1) {
+
+      if (options.skipCompositeChunk()) {
+        // close the request as the single blob. we don't generate the composite blob.
+        blobId = (BlobId) indexToChunkIdsAndChunkSizes.get(0).getFirst();
```

Comment:
Setting blobId to the first chunk ID looks bit hacky :) But, I also don't know any better way.

Reply:
That's true. Added comment to explain it. Hopefully it's less confusing. :-)